libs/js-sdk/src/create-browser-client.ts

```diff
@@ -73,145 +66,173 @@ export function createBrowserClient({
         api || featureBoardHostedService,
     )
 
-    const retryCancellationToken = { cancel: false }
-    retry(async () => {
-        debugLog('SDK connecting in background (%o)', {
-            audiences,
-        })
-        return await updateStrategyImplementation.connect(stateStore)
-    }, retryCancellationToken)
-        .then(() => {
-            if (initialPromise !== initialisedState.initialisedPromise) {
-                return
-            }
+    async function initializeWithAudiences(
+        initializeSpan: Span,
+        audiences: string[],
+    ) {
+        const initialPromise = new PromiseCompletionSource<boolean>()
+        const cancellationToken = { cancel: false }
+        initialPromise.promise.catch(() => {})
+        initialisedState.initialisedPromise = initialPromise
+        initialisedState.initialisedCancellationToken = cancellationToken
+
+        try {
+            await retry(async () => {
+                if (cancellationToken.cancel) {
+                    return
+                }
 
-            if (!initialPromise.completed) {
-                debugLog('SDK connected (%o)', {
-                    audiences,
-                })
-                initialPromise.resolve(true)
-            }
-        })
-        .catch((err) => {
-            if (!initialisedState.initialisedPromise.completed) {
-                debugLog(
-                    'SDK failed to connect (%o): %o',
+                return await tracer.startActiveSpan(
+                    'connect',
                     {
-                        audiences,
+                        attributes: {
+                            audiences,
+                            updateStrategy: updateStrategyImplementation.name,
+                        },
                     },
-                    err,
+                    (connectSpan) =>
+                        updateStrategyImplementation
+                            .connect(stateStore)
+                            .finally(() => connectSpan.end()),
                 )
-                console.error(
-                    'FeatureBoard SDK failed to connect after 5 retries',
-                    err,
+            }, cancellationToken)
+        } catch (error) {
+            if (initialPromise !== initialisedState.initialisedPromise) {
+                addDebugEvent(
+                    "Ignoring initialisation error as it's out of date",
```

Comment:
I'm wondering if event in addDebugEvent() should be a event name and we have a message as an attribute?